lib/drivers/ism43340/ism43340.cpp

```diff
@@ -0,0 +1,270 @@
+#include "Assert.hpp"
+#include "Logger.hpp"
+
+#include <memory>
+
+#define CMD_SET_HUMAN_READABLE "$$$"
+#define CMD_SET_MACHINE_READABLE "---"
+#define CMD_RESET_SETTINGS "Z0"
+
+#define CMD_JOIN_NETWORK "C0"
+#define CMD_SET_SSID "C1="
+#define CMD_SET_PASSWORD "C2="
+#define CMD_SET_SECURITY "C3="
+#define CMD_SET_DHCP "C4="
+#define CMD_GET_CONNECTION_INFO "C?"
+
+#define CMD_SET_TRANSPORT_PROTOCOL "P1="
+#define CMD_SET_HOST_IP "P3="
+#define CMD_SET_PORT "P4="
+#define CMD_START_CLIENT "P6="
+
+#define CMD_SEND_DATA "S3="
+
+Ism43340::Ism43340(SpiPtrT sharedSPI, PinName nCs, PinName intPin,
+                   PinName _nReset)
+    : CommLink(sharedSPI, nCs, intPin), nReset(_nReset) {
+    reset();
+}
+
+inline void Ism43340::formatCommand(std::string* command) {
+    if (command->size() % 2 == 0) {
+        command->append("\r\n");
+    } else {
+        command->append("\r");
+    }
+}
+
+int Ism43340::writetospi(std::string* command) {
+    while (dataready.read() != 1) {
+    }
+
+    uint8_t lastc = 0;
+    radiospi.chipselect();
+    for (int i = 0; i < command->size(); i++) {
+        uint8_t c = command[i];
+        if (lastc != 0) {
+            uint16_t packet = (c << 8) | lastc;
+            radiospi.m_spi->write(packet);
+            lastc = 0;
+        } else {
+            lastc = c;
+        }
+    }
+    radiospi.chipdeselect();
+    return 0;
+}
+
+int Ism43340::readfromspi(uint8* readbuffer, uint32* readlength) {
+    while (dataready.read() != 1) {
+    }
+
+    packetcount = 0;
+    radiospi.chipselect();
+    while (dataready.read() != 0) {
+        uint16_t data = radiospi.m_spi->write(0x0a0a);
+        readbuffer[packetcount] = (uint8_t)(data);
+        readbuffer[packetcount + 1] = (uint8_t)(data >> 8);
+        packetcount += 2;
+    }
+    radiospi.chipdeselect();
+    return 0;
+}
+
+
+/* COMMAND SET DEFINITIONS */
+
+//Check prompt
+inline checkPrompt () {
+    // read the initial prompt into a buffer
+    readfromspi();
+    // check the buffer for the response sequence of the prompt (Not the same as the OK)
+    // you could probably have read spi return an int as a status code for both cant decide
+}
+
+// This command will probably not be used but is here as an example command
+// Set Human Readable
+inline setHumanReadable() {
+    std::string compositeString(CMD_SET_HUMAN_READABLE);
+
+    // formats command adding /r and /n when needed
+    formatCommand(&compositeString);
```

Comment:
Are there any reasons that a command wouldn't have this? It should almost be in a lower function